CMakeLists.txt

```diff
@@ -1387,6 +1393,16 @@ if(HPX_WITH_HWLOC)
   hpx_add_config_define(HPX_HAVE_HWLOC)
 endif()
 
+if(HPX_WITH_STACKOVERFLOW_DETECTION)
+  find_package(LibSigSegv)
+  if(NOT LIBSIGSEGV_FOUND)
+    hpx_error("LibSigSegv could not be found and HPX_WITH_STACKOVERFLOW_DETECTION=ON, please specify LIBSIGSEGV_ROOT to point to the correct location or set HPX_WITH_STACKOVERFLOW_DETECTION to OFF")
+  endif()
+  hpx_libraries(${LIBSIGSEGV_LIBRARIES})
+  include_directories(${LIBSIGSEGV_INCLUDE_DIR})
+  hpx_add_config_define(HPX_HAVE_LIBSIGSEGV)
```

Comment:
Please use the 'same' symbol as you used for the configuration. Normally, for a configuration variable `HPX_WITH_FOO` we use the preprocessor constant `HPX_HAVE_FOO`. In this case this would be `HPX_HAVE_STACKOVERFLOW_DETECTION`.